packet_buffer.go

```diff
@@ -121,19 +124,25 @@ func (pb *packetBuffer) next() (p *Packet, err error) {
 		pb.packetReadBuffer = make([]byte, pb.packetSize)
 	}
 
-	if _, err = io.ReadFull(pb.r, pb.packetReadBuffer); err != nil {
-		if err == io.EOF || err == io.ErrUnexpectedEOF {
-			err = ErrNoMorePackets
-		} else {
-			err = fmt.Errorf("astits: reading %d bytes failed: %w", pb.packetSize, err)
+	// We must return some results to user even if current packet has been filtered
```

Comment:
Could you replace this comment with `// Loop to make sure we return a packet even if first packets are skipped`?

Reply:
Done